crates/cli-support/src/js/mod.rs

```diff
@@ -2082,6 +2076,47 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    fn prestore_global_import_identifiers(&mut self) -> Result<(), Error> {
+        for (_id, adapter) in crate::sorted_iter(&self.wit.adapters) {
+            let instrs = match &adapter.kind {
+                AdapterKind::Import { .. } => continue,
+                AdapterKind::Local { instructions } => instructions,
+            };
+            let mut call = None;
+            for instr in instrs {
+                match instr.instr {
+                    Instruction::CallAdapter(id) => {
+                        if call.is_some() {
+                            continue;
+                        } else {
+                            call = Some(id);
+                        }
+                    }
+                    Instruction::CallExport(_)
+                    | Instruction::CallTableElement(_)
+                    | Instruction::Standard(wit_walrus::Instruction::CallCore(_))
+                    | Instruction::Standard(wit_walrus::Instruction::CallAdapter(_)) => continue,
+                    _ => {}
+                }
+            }
+            if let Some(id) = call {
+                let js = match &self.aux.import_map[&id] {
+                    AuxImport::Value(AuxValue::Bare(js)) => Some(js),
+                    _ => None,
+                };
+                if let Some(js) = js {
+                    match &js.name {
+                        JsImportName::Global { name } => {
+                            self.generate_identifier(name);
```

Comment:
I think this will still want to forward to `import_name` which does caching and error handling, it should just only call that method with `JsImportName::Global` values.

Reply:
sry but I don't quite understand what you suggest by "it should just only call that method with `JsImportName::Global` values." Could you please explain a little bit more?

Reply:
Oh so my point is that I don't think we want to call `generate_identifier` as a bare function since it always generates a new identifier. This means that if the same global is referenced multiple times it'll throw off generation. Instead this should use `import_name` which has error handling and caching which should be used instead of the raw `generate_identifier`